addons/website/static/src/xml/website_form_editor.xml

```diff
@@ -289,13 +289,18 @@
             <t t-call="website.form_field">
                 <select class="form-select s_website_form_input" t-att-name="field.name" t-att-required="field.required || field.modelRequired || None" t-att-id="field.id" style="display: none">
                     <t t-foreach="field.records" t-as="record" t-key="record_index">
-                        <option t-esc="record.display_name" t-att-value="record.id" t-att-selected="record.selected and 'selected' or None"/>
+                        <option t-attf-selected="#{record.selected and 'selected' or None}" t-esc="record.display_name" t-att-value="record.id"/>
```

Comment:
Following the changes introduced in https://github.com/odoo/odoo/commit/6303a3eacdca012649a2ffda627b65c17a7217f1, there is a bug affecting the behavior of t-att-selected, possibly introduced in the subsequent commit: https://github.com/odoo/odoo/commit/363251986dcd753796b4b037a2da2e118b3e8e35. I've created a task (id:3767819). A work around to avoid waiting the forward-port in master, is to use t-attf instead. I could remove this change as soon as the bug is fixed.

Reply:
@Guillaume-gdi I took inspiration from the commit https://github.com/odoo/odoo/commit/b42e9cc686e7d3ccf82cd091a5dc24028fff8a2b by @rdeodoo  to address this issue, which will be resolved in the aforementioned task. 
https://github.com/odoo/odoo/pull/138944/files#diff-e84353750393781d43d4a4bb1debe41db124f3b0ef9301e516f622695905d4b4R292